modules/services/nextdns/default.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nextdns;
+  nextdns = pkgs.nextdns;
+
+in {
+  options = {
+    services.nextdns = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description =
+          "Whether to enable the NextDNS DNS/53 to DoH Proxy service.";
+      };
+      arguments = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "-config" "10.0.3.0/24=abcdef" ];
+        description = "Additional arguments to be passed to nextdns run.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ nextdns ];
+
+    launchd.user.agents.nextdns = {
```

Comment:
Doesn't this need to run as root unless a custom port is specified?

Reply:
Yes it runs on port 53
Do I need to do something special?

When I added that part exactly to my `darwin-configuration` it worked, but if there is another way to test it, I'm interested. Maybe just like home-manager there is a way to specify a darwin repo, they have something like `home-manager -I home-manager=$HOME/Documents/Projects/home-manager switch`

Reply:
In that case this needs to be `launchd.daemons.nextdns`, agents can only run as your local user.

Reply:
thanks! fixed it.